src/TrackGenerator.cpp

```diff
@@ -423,13 +406,27 @@ FP_PRECISION TrackGenerator::getMaxOpticalLength() {
           sigma_t = material->getSigmaT();
 
           for (int e=0; e < material->getNumEnergyGroups(); e++)
-            max_optical_length = std::max(max_optical_length, length*sigma_t[e]);
+            max_optical_length = std::max(max_optical_length,
```

Comment:
And can you simply update `_max_optical_length` here directly and remove line 416-417 below for simplicity?